From cb860ab6fba86f646bcd3bff6cadcc2bdd3c6964 Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Sat, 4 Nov 2017 21:58:10 +0200
Subject: [PATCH] build: enable pseh only in x86

pseh supports only x86, no point in enabling it when libraries are
enabled. This enables downstream to enable libraries without failing.

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
---
 configure.ac | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 468d1b19..cfb07362 100644
--- a/configure.ac
+++ b/configure.ac
@@ -58,10 +58,14 @@ AC_ARG_WITH([libraries],
   [with_libraries=no])
 AS_CASE([$with_libraries],
   [yes|all],[
-    with_libraries="libmangle,pseh,winpthreads"
+    with_libraries="libmangle,winpthreads"
     with_libraries_winpthreads=yes
     with_libraries_mangle=yes
-    with_libraries_pseh=yes],
+    AS_CASE([$host_cpu],
+      [i?86], [
+        with_libraries="$with_libraries,pseh"
+        with_libraries_pseh=yes
+      ])],
   [libmangle],[
     with_libraries="libmangle"
     with_libraries_winpthreads=no
-- 
2.13.6

